public/scripts/main.js

```diff
@@ -524,53 +565,82 @@ speechModal.addEventListener('click', function() {
       speechTextArea.textContent += phrases[0];
     }
   })
-})
-
-var createSpeechForm = function() {
-  var divForm = document.createElement('div');
-  divForm.className = 'div-form';
-  var textForm = document.createElement('form');
-
-  var textArea = document.createElement('textarea');
-  textArea.setAttribute('id', 'speech-form');
-  textArea.setAttribute('name', 'addText');
-  textArea.setAttribute('placeholder', 'Say something...')
-  textArea.className = 'speech-form';
-  textArea.classList.add('materialize-textarea');
+});
 
-  var titleArea = createTitleInput()
+//###########################
 
-  var progressBar = createProgressBar();
+let createVideoObj = function() {
+  var videoForm = document.querySelector('form')
+  var url = videoForm.videoUrl.value;
+  var filePath = parseYoutube(url);
+  var embedUrl = `https://www.youtube.com/embed/${filePath}?rel=0`
+  var title = videoForm.addTitle.value;
+  var videoObj = {
+    type: 'video',
+    title: title,
+    content: embedUrl,
+    order: getOrder(),
+  }
+//  order += 1;
+//  displayedThoughts.push(videoObj);
+  return videoObj;
+};
 
-  textForm.appendChild(titleArea);
-  textForm.appendChild(progressBar);
-  textForm.appendChild(textArea);
-  divForm.appendChild(textForm);
+let createVideoElements = function(videoObj) {
```

Comment:
Do you think it would make sense to create a method to create a template for image/video card, since they're so similar? It can cause some issues accessing some elements to name their classes though...